src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -127,7 +133,26 @@ public static ContentSource newMarkLogicContentSource(MarkLogicRepositoryDescrip
         String dbname = StringUtils.defaultIfBlank(descriptor.dbname, DB_DEFAULT);
         String user = descriptor.user;
         String password = descriptor.password;
-        return ContentSourceFactory.newContentSource(host, port.intValue(), user, password, dbname);
+        // handle SSL
+        SecurityOptions securityOptions = null;
+        if (descriptor.sslEnabled) {
+            securityOptions = newTrustOptions();
+        }
+        return ContentSourceFactory.newContentSource(host, port.intValue(), user, password, dbname, securityOptions);
+    }
+
+    protected static SecurityOptions newTrustOptions() {
+        try {
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("SunX509");
+            // init trust manager factory with a null KeyStore to use the default one
+            trustManagerFactory.init((KeyStore) null);
+            TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+            SSLContext sslContext = SSLContext.getInstance("SSLv3");
```

Comment:
`SSLv3` is obsolete and has security issues, it would be better to use the latest, `TLSv1.2`. But if `SSLContext.setDefault()` works, even better, let's use it to avoid hardcoding protocols that may evolve.